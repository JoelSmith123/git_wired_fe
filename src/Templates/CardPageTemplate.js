import React, { Component } from 'react';
import './Templates.css';

export default class CardPageTemplate extends Component {
  constructor() {
    super()
  }

  render() {
    return (
      <div className='CardPageTemplate'>
        <div className='template-header'></div>
        <div className='template-card-container'>
          <div className='template-card'></div>
          <div className='template-card'></div>
          <div className='template-card'></div>
          <div className='template-card'></div>
        </div>
        <div className='template-footer'></div>
      </div>
    )
  }
}