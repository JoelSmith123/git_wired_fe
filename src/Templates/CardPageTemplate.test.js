import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import CardPageTemplate from './CardPageTemplate.js'

describe('CardPageTemplate', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CardPageTemplate />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<CardPageTemplate />)
    expect(wrapper).toMatchSnapshot()
  })   
})