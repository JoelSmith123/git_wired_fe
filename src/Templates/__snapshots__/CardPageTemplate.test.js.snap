// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPageTemplate should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardPageTemplate />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="template-header"
        >
          <div
            className="pseudo-btn-space"
          />
          <i
            className="far fa-plus-square"
            onClick={[Function]}
          />
        </div>,
        <div
          className="template-card-container"
        >
          <div
            className="template-card"
          />
          <div
            className="template-card"
          />
          <div
            className="template-card"
          />
          <div
            className="template-card"
          />
        </div>,
        <div
          className="template-footer"
        />,
      ],
      "className": "CardPageTemplate",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="pseudo-btn-space"
            />,
            <i
              className="far fa-plus-square"
              onClick={[Function]}
            />,
            Array [],
          ],
          "className": "template-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "pseudo-btn-space",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "far fa-plus-square",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="template-card"
            />,
            <div
              className="template-card"
            />,
            <div
              className="template-card"
            />,
            <div
              className="template-card"
            />,
          ],
          "className": "template-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "template-card",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "template-card",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "template-card",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "template-card",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "template-footer",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="template-header"
          >
            <div
              className="pseudo-btn-space"
            />
            <i
              className="far fa-plus-square"
              onClick={[Function]}
            />
          </div>,
          <div
            className="template-card-container"
          >
            <div
              className="template-card"
            />
            <div
              className="template-card"
            />
            <div
              className="template-card"
            />
            <div
              className="template-card"
            />
          </div>,
          <div
            className="template-footer"
          />,
        ],
        "className": "CardPageTemplate",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="pseudo-btn-space"
              />,
              <i
                className="far fa-plus-square"
                onClick={[Function]}
              />,
              Array [],
            ],
            "className": "template-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "pseudo-btn-space",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "far fa-plus-square",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="template-card"
              />,
              <div
                className="template-card"
              />,
              <div
                className="template-card"
              />,
              <div
                className="template-card"
              />,
            ],
            "className": "template-card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "template-card",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "template-card",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "template-card",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "template-card",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "template-footer",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
